apis/python/src/tiledbsoma/soma_collection.py

```diff
@@ -11,8 +21,11 @@
 from .tiledb_object import TileDBObject
 from .tiledb_platform_config import TileDBPlatformConfig
 
+# A collection can hold any sub-type of TileDBObject
+CollectionElementType = TypeVar("CollectionElementType", bound=TileDBObject)
 
-class SOMACollection(TileDBObject):
+
```

Comment:
maybe add a quick comment what is the value-add over `SOMACollection`

Reply:
there is no longer a SOMACollection other than this.  There is just SOMACollectionBase

This one implements the general ability to store SOMA objects.  The generic allows type constrains, such as we require in SOMAExperiment and SOMAMeasurement

Reply:
comment added

Reply:
ahhhh ok thanks!